airflow_adapter.py

```diff
@@ -0,0 +1,119 @@
+"""The Airflow Adapter."""
+from adapter_abc import Adapter
+from filter_releases import filter_releases
+from filter_releases import sort
+from from_sql import read_releases
+from from_sql import read_tasks
+from to_sql import to_sql_release
+from to_sql import to_sql_releases
+from to_sql import to_sql_task
+from to_sql import to_sql_tasks
+
+
+class AirflowAdapter(Adapter):
+  """Provides a way to interact with the Airflow Database, and get data from it."""
+
+  def __init__(self, airflow_db):
+    self._airflow_db = airflow_db
+
+    self._releases = self._airflow_db.query('SELECT * FROM dag_run;')
+    self._releases = read_releases(self._releases, self._airflow_db)
+
+    branches = set()
+    for release in self._releases:
+      branches.add(self._releases[release].branch)
+    self._branches = list(branches)
+
+    types = set()
+    for release in self._releases:
+      types.add(self._releases[release].release_type)
+    self._types = list(types)
+
+  def get_releases(self, start_date, end_date, datetype, state,
```

Comment:
I think passing a filter object would be easier to use here since we are updating the interface.

Reply:
So just to check, the filter object should function in a similar manner to the release object, and then the filter conditions are called like this:
```
start_date = filter.start_date
datetype = filter.datetype
```
?

Reply:
Yes. I would call it filter_options.